### PR TITLE
Add all polynomial options to RBF Eigen mappings

### PR DESCRIPTION
**Pull request by DavidSCN**
## Main changes of this PR
Replaces #1166, which was too stale in order to recycle it. Related to #1162.

## Motivation and additional information
Depends on #1320. This gives more flexibility for our RBF mappings and how to treat the resulting systems.

## Author's checklist

* [x] I added a changelog file with `make changelog` if there are user-observable changes since the last release.
* [x] I ran `make format` to ensure everything is formatted correctly.
* [x] I sticked to C++14 features.
* [x] I sticked to CMake version 3.16.3.
* [x] I squashed / am about to squash all commits that should be seen as one.
* [x] Add proper testing for the polynomial options

## Reviewers' checklist

* [ ] Does the changelog entry make sense? Is it formatted correctly?
* [ ] Do you understand the code changes?

### CODE REVIEW

File: src/mapping/RadialBasisFctMapping.hpp [hunk 1]
```diff
@@ -212,7 +219,7 @@ void RadialBasisFctMapping<RADIAL_BASIS_FUNCTION_T>::mapConservative(DataID inpu
 
     // Construct Eigen vectors
     Eigen::Map<Eigen::VectorXd> inputValues(globalInValues.data(), globalInValues.size());
-    Eigen::VectorXd             outputValues((_rbfSolver.getEvaluationMatrix().cols() - this->getPolynomialParameters()) * valueDim);
+    Eigen::VectorXd             outputValues((this->output()->getGlobalNumberOfVertices()) * valueDim);
```
Comment (DavidSCN) [inline src/mapping/RadialBasisFctMapping.hpp:222]: I was not sure whether it is a good idea to rely on `getGlobalNumberOfVertices`. I needed to add the information in all places in the RBF unit tests. Motivation was here to get rid of the dependency of the polynomial.

Comment (uekerman) [reply src/mapping/RadialBasisFctMapping.hpp:222]: It should give the correct value AFAIK. Alternative could be to store this in `_rbfSolver`.

**Review by uekerman — APPROVED** [review]
Looks good :+1: Did not check every detail, but I understand the code overall.

What do we do with "polynomial = off"? Is it allowed? It seems we have no unit tests for this option.

The class `RadialBasisFctSolver` needs a better name or at least some minimal class documentation. At first sight, it is not clear that this class only does direct solving and has nothing to do with `PetRadialBasisFctMapping`.

**Comment by DavidSCN** [comment]
> What do we do with "polynomial = off"? Is it allowed? It seems we have no unit tests for this option.

I need to add more tests here anyway. `polynomial=off` works, but it doesn't result in the same accuracy for our simple test cases, so I still need to set up a few new test cases to cover this option.

**Pull request merged**